src/app/games/go/tests/GoGroupDatas.spec.ts

```diff
@@ -20,6 +20,23 @@ describe('GoGroupDatas:', () => {
                                                      []);
         expect(() => group.getWrapper()).toThrowError(`Can't call getWrapper on non-mono-wrapped group`);
     });
+    it('should throw when addPawn is called two times with the same coord', () => {
+        // Givne any GoGroupDatas containing "coord" already
+        const group: GoGroupDatas = new GoGroupDatas(GoPiece.EMPTY, [], [], [], [], []);
+        group.addPawn(coord, GoPiece.BLACK);
+
+        // When adding the coord again
+        // Then it should throw
+        const expectedError: string = 'Assertion failure: This group already contains (0, 0) (extra data: undefined)';
+        let failed: boolean = false;
+        try {
+            group.addPawn(coord, GoPiece.BLACK);
+        } catch (error) {
+            expect(error.message).toBe(expectedError);
+            failed = true;
+        }
+        expect(failed).toBeTrue();
```

Comment:
ah non hein pas cette fois-ci :D expect(() => group.addPawn).toThrowError(expectedError) !

Reply:
Bha, il a des comportements randomisés, des fois il marche des fois pas !

Reply:
eeuh mais est-ce qu'on souhaite ces comportements randomisés ? Si oui, ça me semble louche ? Si non, alors faut fixer le comportement

Reply:
J'ai constaté que si ErrorLogger est testé avant ou après cause la randomité, mais pas plus de détail, c'est malheureusement pas [que je sache] faisable de mettre des ordres d'executions des tests, ne fut-ce que temporairement.

Reply:
faut l'mocker !
`        spyOn(ErrorLoggerService, 'logError').and.callFake(ErrorLoggerServiceMock.logError);`